### PR TITLE
fix: iOS primitive rendering issue

### PR DESCRIPTION
**Pull request by pramodcog**

### CODE REVIEW

File: viewer/packages/rendering/src/glsl/sector/primitives/cone.frag [hunk 1]
```diff
@@ -156,7 +157,6 @@ void main() {
       }
   #endif
 
-
-    float fragDepth = updateFragmentDepth(p, projectionMatrix);
-    updateFragmentColor(renderMode, color, v_treeIndex, normal, fragDepth, matCapTexture, GeometryType.Primitive);
+    updateFragmentDepth(p, v_projectionMatrix);
+    updateFragmentColor(renderMode, color, v_treeIndex, normal, gl_FragCoord.z, matCapTexture, GeometryType.Primitive);
```
Comment (larsmoa) [inline viewer/packages/rendering/src/glsl/sector/primitives/cone.frag:161]: Why use gl_FragCoord.z?

Comment (pramodcog) [reply viewer/packages/rendering/src/glsl/sector/primitives/cone.frag]: This is wrong, I think, I pushed while testing/identifying the issue, my bad!

**Comment by hayer** [comment]
LGTM. Important fix for my team. @larsmoa could you please prioritize reviewing this? 👼🏻

**Review by larsmoa — CHANGES_REQUESTED** [review]
Add `flat` for all of the "uniforms"

**Comment by pramodcog** [comment]
> Add `flat` for all of the "uniforms"

I doubt that we could not use `flat` with uniform, let me try and check!

**Comment by larsmoa** [comment]
> > Add `flat` for all of the "uniforms"
> 
> I doubt that we could not use `flat` with uniform, let me try and check!
> 
> Yup, it is not valid. Uniforms does not take `flat` qualifier

I mean that use `flat` for all the varyings that replace uniforms. Flat for uniforms doesnt make sense as its constant.

**Review by larsmoa — CHANGES_REQUESTED** [review]
As mentioned previously, we have a lot of uniforms in many fragment shader - have you assessed what works and what doesn't? We need to be thorough on this!

File: viewer/packages/rendering/src/glsl/sector/primitives/cone.frag [hunk 2]
```diff
@@ -125,13 +117,17 @@ void main() {
...
       ) {
         // Missed the other point too
         discard;
       }
     }
+    #if NUM_CLIPPING_PLANES > 0
+      if (isClipped(p, v_clippingPlanes)) {
+        discard;
+      }
+    #endif
```
Comment (larsmoa) [inline viewer/packages/rendering/src/glsl/sector/primitives/cone.frag:130]: Why is this moved?

Comment (pramodcog) [reply viewer/packages/rendering/src/glsl/sector/primitives/cone.frag:130]: `NUM_CLIPPING_PLANES` value is assigned by the threejs only when clipping is active, else its value is always `0`, so the earlier, the check for discarding the clipped pixel function call will have an compiler error as I am passing `v_clippingPlanes[NUM_CLIPPING_PLANES ]` as a parameter.
